Index: xserver/hw/kdrive/src/kinput.c
===================================================================
--- xserver.orig/hw/kdrive/src/kinput.c	2005-06-17 14:10:03.000000000 +0000
+++ xserver/hw/kdrive/src/kinput.c	2005-06-17 14:04:54.000000000 +0000
@@ -326,8 +326,8 @@
     case DEVICE_ON:
 	pDev->on = TRUE;
 	pKdPointer = pDevice;
-	for (i = 0; i < kdNMouseFuncs; i++)
-	    (*kdMouseFuncs[i]->Init)();
+	for (i = kdNMouseFuncs; i > 0 ; i--)
+	    (*kdMouseFuncs[i-1]->Init)();
 	break;
     case DEVICE_OFF:
     case DEVICE_CLOSE:
